.github/workflows/ci.yml

```diff
@@ -731,25 +728,36 @@ jobs:
         run: |
           conda info
           conda list
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_binary.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_default.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_indexing.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_iter.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_join_sort.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_map_metadata.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_pickle.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_reduce.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_udf.py
-      - run: pytest -n 2 modin/pandas/test/dataframe/test_window.py
-      - run: pytest -n 2 modin/pandas/test/test_api.py
-      - run: pytest -n 2 modin/pandas/test/test_concat.py
+      - name: Setup parallel jobs env
+        run: echo "PYTEST_DIST_WORKERS=2" >> $GITHUB_ENV
+        if: matrix.engine != 'ray'
+      - name: Setup parallel jobs env
+        run: echo "PYTEST_DIST_WORKERS=1" >> $GITHUB_ENV
+        if: matrix.engine == 'ray'
+      - name: Start local ray cluster
+        run: |
+          ray start --head --port=6379 --object-store-memory=1000000000
+          echo "MODIN_RAY_CLUSTER=True" >> $GITHUB_ENV
```

Comment:
I thought ray automatically detects the existing cluster. Am I incorrect?

Reply:
When I looked at `initialize_ray` I saw that it checked for `override_is_cluster` or the env variable for `IsRayCluster` which had to be set with `MODIN_RAY_CLUSTER`. I tried testing this locally as well.

Reply:
Ah, it seems that to attach to an existing ray cluster without `MODIN_RAY_CLUSTER`, you have to run `ray.init` within the same process, e.g. [here](https://modin.readthedocs.io/en/latest/getting_started/using_modin/using_modin_cluster.html#connecting-to-a-ray-cluster).